src/screens/SignMessage/03-ConnectDevice.js

```diff
@@ -0,0 +1,146 @@
+// @flow
+import invariant from "invariant";
+import React, { useState, useMemo, useEffect, useCallback } from "react";
+import { StyleSheet } from "react-native";
+import { useSelector } from "react-redux";
+import { from } from "rxjs";
+import SafeAreaView from "react-native-safe-area-view";
+import { createAction as createAppAction } from "@ledgerhq/live-common/lib/hw/actions/app";
+import type { TypedMessageData } from "@ledgerhq/live-common/lib/families/ethereum/types";
+import type { MessageData } from "@ledgerhq/live-common/lib/hw/signMessage/types";
+import connectApp from "@ledgerhq/live-common/lib/hw/connectApp";
+import { withDevice } from "@ledgerhq/live-common/lib/hw/deviceAccess";
+import signMessage from "@ledgerhq/live-common/lib/hw/signMessage";
+import { accountScreenSelector } from "../../reducers/accounts";
+import DeviceAction from "../../components/DeviceAction";
+import { TrackScreen } from "../../analytics";
+import { ScreenName } from "../../const";
+
+const initialState = {
+  signMessageRequested: null,
+  signMessageError: null,
+  signMessageResult: null,
+};
+const createAction = connectAppExec => {
```

Comment:
should this be a new "action" to be put in live-common?

Reply:
yes I think so, I imagine it will be reused by desktop